src/registry.rs

```diff
@@ -20,33 +21,62 @@ pub trait RegistryObserver {
     );
 }
 
+#[derive(Debug, Clone)]
+pub struct Config {
+    entry_ttl: Duration,
+}
+
+impl Config {
+    pub fn entry_ttl(&self) -> Duration {
+        self.entry_ttl
+    }
+
+    pub fn set_entry_ttl(self, value: Duration) -> Self {
+        Self { entry_ttl: value }
+    }
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self { entry_ttl: Duration::from_secs(15) }
+    }
+}
+
 #[derive(Clone, Debug)]
 pub struct Registry<T: RegistryObserver> {
     external_services_by_intent: HashMap<IntentConfiguration, HashSet<ServiceConfiguration>>,
-    known_services: HashSet<ServiceConfiguration>,
+    known_services: HashMap<ServiceConfiguration, SystemTime>,
```

Comment:
for my understanding: I didn't see a lookup on the time by configuration. Something we plan to add?

Reply:
No, there's no look-up. The code does a scan of the map during pruning. `ServiceConfiguration` serves as a key for `HashSet` so one can't put the timestamp in there (besides it not being configuration data). So yes, while the `HashMap` may mislead to think there's some sort of look-up being done, it's really used to pair the variant data together with the invariant part. The other option would be to keep a list _and_ a set of service keys. It would add some overhead without any benefit.

Reply:
Makes sense. It was a bit confusing as I thought we do ttl lookup by ServiceConfiguration.